x/model/keeper/msg_server_model_version.go

```diff
@@ -135,3 +135,55 @@ func (k msgServer) UpdateModelVersion(goCtx context.Context, msg *types.MsgUpdat
 
 	return &types.MsgUpdateModelVersionResponse{}, nil
 }
+
+func (k msgServer) DeleteModelVersion(goCtx context.Context, msg *types.MsgDeleteModelVersion) (*types.MsgDeleteModelVersionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check signer has enough rights to delete model version
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Address: (%s)", err)
+	}
+	if err := checkModelRights(ctx, k.Keeper, signerAddr, msg.Vid, "MsgDeleteModelVersion"); err != nil {
+		return nil, err
+	}
+
+	// check if model version exists
+	modelVersion, isFound := k.GetModelVersion(
+		ctx,
+		msg.Vid,
+		msg.Pid,
+		msg.SoftwareVersion,
+	)
+
+	if !isFound {
+		return nil, types.NewErrModelVersionDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion)
+	}
+
+	if msg.Creator != modelVersion.Creator {
```

Comment:
This check is wrong. `checkModelRights` is enough. A different Vendor account with the same VID as the account who created the model (model version), can update/delete this model (model version).
Please add a corresponding test to make sure it's true.

Reply:
Fixed, added test